### PR TITLE
feat: Universal Trials Mode

### PR DESCRIPTION
**Pull request by two4teezee**
A massive PR that basically takes this: https://github.com/two4teezee/Ikemen-GO-Trials-Mode and merges it into the core engine. Includes all necessary changes to the default config file, system.base.def, etc. Unsurprisingly, given how extensive the mode is, it touches many aspects of the existing Lua files. The only real gains from delivering the mode with the engine would be...

1) Greater visibility and accessibility to the mode
2) A small (extremely small) efficiency gained by not having to open the char .def files in Lua after the engine has already parsed them.
3) No duplicate code between training.zss and trials.zss

### CODE REVIEW

**Comment by two4teezee** [comment]
If this gets merged in, should also merge this: https://github.com/ikemen-engine/Ikemen_GO-Elecbyte-Screenpack/pull/20

File: external/script/trials.lua [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+		end
+	end
+end
+
+-- Find trials files and parse them; append t_selChars table
+function trials.f_parseTrials(row)
+	i = 0 --Trial number
+	j = 0 --TrialStep number
+	trial = {}
+	local trialsFile = main.f_fileRead(main.t_selChars[row].trialspath, "r")
```
Comment (bluish-elvis) [inline external/script/trials.lua]: ~~If Character has not trials.def, IKEMEN will crash due to not found trials.def.
It looks like necessary to check it by main.f_fileExists~~

Comment (two4teezee) [reply external/script/trials.lua]: You aren't wrong but this function is only invoked by start.lua if trialspath ~= "" - in other words, it's only resolved when a trials file was found when reading the char def file.

Comment (bluish-elvis) [reply external/script/trials.lua:785]: also I tested this build and your sample of KFM trials.def but panicError caused on this f_fileRead even there IS trials.def...
```suggestion
	local trialsFile = main.f_fileRead(main.t_selChars[row].dir .. main.t_selChars[row].trialspath, "r")
```

Comment (bluish-elvis) [inline external/script/trials.lua]: also I tested this build and your sample of KFM trials.def but panicError thrown by this f_fileRead even there IS trials.def...
```suggestion
	local trialsFile = main.f_fileRead(main.t_selChars[row].dir .. main.t_selChars[row].trialspath, "r")
```

Comment (two4teezee) [reply external/script/trials.lua]: Thanks, it resolved the path in MacOS, but you're right, what I had wasn't working in x64. I accepted the change, try it again @bluish-elvis

Comment (two4teezee) [reply external/script/trials.lua]: This change was breaking for MacOS but I included a change that leveraged your first suggestion as well.

Comment (bluish-elvis) [reply external/script/trials.lua]: Confirmed it's work. thanks!

**Pull request closed without merging**